core-to-plc/src/Language/Plutus/CoreToPLC.hs

```diff
@@ -226,8 +229,10 @@ convTyConApp tc ts
     | tc == GHC.intPrimTyCon = pure $ appSize haskellIntSize (PLC.TyBuiltin () PLC.TyInteger)
     -- we don't support Integer
     | GHC.tyConName tc == GHC.integerTyConName = unsupported "Integer: use Int instead"
-    -- this is Void#, goes to 'forall a. a'
-    | tc == GHC.voidPrimTyCon = safeFreshTyName "any" >>= \n -> pure $ PLC.TyForall () n (PLC.Type ()) (PLC.TyVar () n)
+    -- this is Void#, see Note [Value restriction]
+    | tc == GHC.voidPrimTyCon = do
+          tyname <- safeFreshTyName "a"
+          mangleTyForall $ PLC.TyForall () tyname (PLC.Type ()) (PLC.TyVar () tyname)
```

Comment:
I think, it would be better to give that type a name (like `errorFunc` below) and use the name here as it would read better.